File: server/src/main/java/org/dan/jadalnia/app/festival/order/pojo/Oid.java
package org.dan.jadalnia.app.festival.order.pojo;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.dan.jadalnia.sys.type.number.ImmutableNumber;

public class Oid extends ImmutableNumber {
    @JsonCreator
    public Oid(int id) {
        super(id);
    }

    // jax-rsp
    @JsonCreator
    public static Oid valueOf(String s) {
        return new Oid(Integer.valueOf(s));
    }

    public static Oid of(int id) {
        return new Oid(id);
    }
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/order/pojo/Oid.kt
package org.dan.jadalnia.app.order.pojo;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.dan.jadalnia.sys.type.number.ImmutableNumber;

class Oid @JsonCreator constructor (id: Int): ImmutableNumber(id) {
    companion object {
        // jax-rsp
        @JsonCreator
        fun valueOf(s: String): Oid = Oid(Integer.valueOf(s))

        fun of(id: Int): Oid = Oid(id)
    }
}
